x-pack/plugins/data_enhanced/server/search/session/session_service.ts

```diff
@@ -79,11 +71,21 @@ export class SearchSessionService implements ISessionService {
   private sessionSearchMap = new Map<string, SessionInfo>();
   private internalSavedObjectsClient!: SavedObjectsClientContract;
   private monitorTimer!: NodeJS.Timeout;
+  private config!: ConfigSchema['search']['sessions'];
 
-  constructor(private readonly logger: Logger) {}
+  constructor(private readonly logger: Logger, private readonly config$: Observable<ConfigSchema>) {
+    (async () => {
+      const config = await this.config$.pipe(first()).toPromise();
+      this.config = config.search.sessions;
+    })();
```

Comment:
this looks scary, I'd use the older way where we await this promise in a method before using the `config`.
I think we shouldn't rely on that, for example, `setupMonitoring` is called only after this promise is resolved

Reply:
We have this piece of code in a lot of places in our code base, including in `core` code.
Otherwise, how can I extract config asynchronously in the synchronous `setup` \ `start` life-cycle methods?

Reply:
Hmm.. looks like I just haven't bumped into doing it like this.

I know we sometimes use this [helper](https://github.com/elastic/kibana/blob/master/src/plugins/kibana_utils/public/core/create_start_service_getter.ts) in case we really need access to start services without `await`. But at least it will fire a proper error message and not just `this.config is undefined` in case used incorrectly.

Reply:
@lukasolson @Dosant so I've improved the implementation, but it still uses the fact that the session service's `start` method is `async` (unlike the real plugin interface).

If I were to do any other implementation, it would be hard to use the `config` object in the default params of the `save` method.

Hope it's good with you :)